src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -58,12 +66,14 @@ public async Task<MsalTokenResponse> GetTokenAsync(Uri endPoint, RequestContext
             return await ExecuteRequestAsync<MsalTokenResponse>(endPoint, HttpMethod.Post, requestContext).ConfigureAwait(false);
         }
 
-
-
-        internal async Task<T> ExecuteRequestAsync<T>(Uri endPoint, HttpMethod method, RequestContext requestContext, bool expectErrorsOn200OK = false)
+        internal async Task<T> ExecuteRequestAsync<T>(Uri endPoint, HttpMethod method, RequestContext requestContext, bool expectErrorsOn200OK = false, bool respondToDeviceAuthChallenge = true)
         {
             bool addCorrelationId = requestContext != null && !string.IsNullOrEmpty(requestContext.Logger.CorrelationId.ToString());
-            AddCommonHeaders(requestContext, addCorrelationId);
```

Comment:
I think this is wrong, but please correct me if my judgement fails - we should ALWAYS add the correlation ID. You probably want to prevent only the PKAuth header from being added.

Reply:
Yes, we will always send the correlation id. What is happening here is the pkeyauth code is replaying the same request after being issued a challenge with additional pkeyauth headers. We don't want to re-add common headers to the request because the common headers have already been added to it the first time